### PR TITLE
Seed events

### PR DESCRIPTION
**Pull request by martinothamar**
This will seed 6 events if there are none in the DB. Requested by @anhtin

### CODE REVIEW

**Comment by raymond-selvik** [comment]
Good idea, but will not the seeder be injected in every environment now? We would only be interested to seed the database in local debug I guess? I would rather prefer a script to init the database, rather than implementing in code, when it should only used by one environment. 

Perhaps we can specify only to inject when the environment is local?

**Comment by raymond-selvik** [comment]
And if we go for the as-code way, I think it maybe could be an idea to avoid hardcoding seed data in the code. And perhaps extract it from a JSON-file, that we can quickly modify for our needs.

**Comment by martinothamar** [comment]
I can make sure it only executes in `Development` environment or add an environment-flag (`ENABLE_SEEDING` = `true` | `false`).
The benefit of doing it in code is we are forced to update the seeding code when the model changes (strong typing, compile time errors). Is JSON really much easier to edit than the code?

File: src/MadLearning/MadLearning.API.Infrastructure/Persistence/EventRepository.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ public async Task<List<EventModel>> GetEvents(EventFilterApiDto eventFilter, Can
         {
             var filter = CreateFilter(eventFilter);
 
-            var events = await this.collection.FindAsync(filter, cancellationToken: cancellationToken);
+            var events = this.collection.Find(filter).Limit(eventFilter.Limit);
```
Comment (martinothamar) [inline src/MadLearning/MadLearning.API.Infrastructure/Persistence/EventRepository.cs:42]: Just to make sure, there is no blocking on async IO here, if I read this right: https://github.com/mongodb/mongo-csharp-driver/blob/b9b4d66003bf49a364117b395b3f710b3f4c83d4/src/MongoDB.Driver/IMongoCollectionExtensions.cs#L822

**Comment by raymond-selvik** [comment]
> I can make sure it only executes in `Development` environment or add an environment-flag (`ENABLE_SEEDING` = `true` | `false`).
> The benefit of doing it in code is we are forced to update the seeding code when the model changes (strong typing, compile time errors). Is JSON really much easier to edit than the code?

I agree with the arguments that is force us us to update the seeding code if the model changes. But my view is that configuration, which I kind of think seeding data is, should be decoupled from the logic of the code. 

But maybe @LERoise has some words in this case.

**Review by LERoise — APPROVED** [review]
I dont see any big problems either way, only difference is control/timing of execution.
Depending on the amount of data and goal of the seeding, but is something that needs to be considered each time.

But this is ok, but i would consider moving it to a different service so it can be called more on demand rather then always(only in dev)

**Review by anhtin — APPROVED** [review]

**Pull request merged**